src/weather/tests/weather.service.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { WeatherService } from '../weather.service';
+import { BadRequestException } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { ConfigService } from '@nestjs/config';
+import { of } from 'rxjs';
+import { WEATHER_ERROR } from '../../errors/weather.errors';
+
+describe('WeatherService', () => {
+  let weatherService: WeatherService;
+
+  beforeEach(async () => {
```

Comment:
This doesn't need to be run before each test, it can be run once for the whole suite

```suggestion
  beforeAll(async () => {
```